### PR TITLE
Ek 406 update gatsby to v3

### PR DESCRIPTION
**Pull request by cs-paz**
Summary of Changes:

1. Added a `.nvmrc` file and set the version to `14.17.4` (latest stable version).
2. Added `@link` extensions in our schema for the array fields. Resolvers are no longer inferred by default in gatsby v3.
3. Removed createRemoteFileNode.
4. Migrated from using `gatsby-image` to `gatsby-plugin-image` for an updated API for v3.  Removed `gatsby-image` package.  
5. Deleted unused `Image.js` component and updated `Index.js` to not contain all of gatsby's default content that we never took out.

Test:
1. There should be no noticeable changes from what is currently in prod and this build.
2. All array fields are being rendered correctly (this goes with test #1 but this was a major pain point so better to note it in tests)
3. Test caching:
- Deploy with cache clear on Netlify
- Change one of the photos `last_modified` filed on `people-dev2`.
- Deploy without clearing cache
- Check the build log to see if the modified photo is being downloaded again and all others are imported from the cache.

Notes:
1. Remove these two lines `[context.production]
    environment = { PEOPLEAPI_HOST = 'api.stevens.edu/people-dev2/faculty'}` from `netlify.toml` after testing is done.
2. Merge https://bitbucket.org/stevens_devops/people/pull-requests/46/ek-431-tweak-people-api before merging this PR because there are changes here that depend on the changes in the People API.
3. Photos do not work on Cloudflare at the moment and that's because we're using Netlify caching plugin.

### CODE REVIEW

File: gatsby_fp/gatsby-config.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 require("dotenv").config()
+process.env.GATSBY_CONCURRENT_DOWNLOAD = '50'
```
Comment (mf-stevens) [inline gatsby_fp/gatsby-config.js:2]: I already reviewed this change, so it should not be up for review a second time. I think if you merge master into this branch, it'll clear this up.

**Review by mf-stevens — CHANGES_REQUESTED** [review]
When comparing https://faculty.stevens.edu/kabel against https://48ac49a3.faculty-profiles.pages.dev/kabel there are 2 concerns:
1. In Chrome and Firefox, the photo only shows up on prod, not the branch preview. (In Safari, it shows up in the branch preview as well, oddly enough.)
2. The console (in browser developer tools) displays no errors in prod, but it displays `[gatsby-plugin-image] You're missing out on some cool performance features. Please add "gatsby-plugin-image" to your gatsby-config.js` due to these changes. Can we implement the advice it's giving us?

File: gatsby_fp/gatsby-config.js (hunk 1 above)
Comment (cs-paz) [reply gatsby_fp/gatsby-config.js]: Removed this line of code and merged master

File: gatsby_fp/netlify.toml [hunk 2]
```diff
@@ -1,2 +1,2 @@
 [context.deploy-preview.environment]
-    PEOPLEAPI_HOST = 'api.stevens.edu/people/faculty'
+    PEOPLEAPI_HOST = 'api.stevens.edu/people-dev2/faculty'
```
Comment (dmotan) [inline gatsby_fp/netlify.toml]: Why would we use people-dev2 here?

Comment (cs-paz) [reply gatsby_fp/netlify.toml]: This was made temporarily while we were testing on people-dev2. I was planning on changing this back upon approval of EK-428. @dmotan

Comment (dmotan) [reply gatsby_fp/netlify.toml]: I see, I believe this should be fixed as am not sure if we want to continue with EK-428.

File: gatsby_fp/.nvmrc [hunk 3]
```diff
@@ -0,0 +1 @@
+14.17.0
```
Comment (dmotan) [inline gatsby_fp/.nvmrc]: Why would we use node 14.17.0 and not the stable version 14.1.0?
https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/#minimal-nodejs-version-12130

Comment (mf-stevens) [reply gatsby_fp/.nvmrc]: Please provide a link to something that says 14.1.0 is more stable than 14.17.0.

Based on https://github.com/nodejs/Release#nodejs-release-working-group it seems like "14.x" is active LTS and typically "x" means the highest number available.

Comment (dmotan) [reply gatsby_fp/.nvmrc]: Good catch. For some reason, when I ran `nvm list` on my local machine, I got `stable -> 14.1 (-> v14.1.0) (default)`.
But https://nodejs.org/en/ is where I should have looked first, which shows 14.7.4.

Comment (dmotan) [reply gatsby_fp/.nvmrc]: Or I should've run `nvm ls-remote` which shows the list of available node versions on remote.

**Comment by dmotan** [comment]
https://build-70806b1a-8a34-40a0-bbf5-89d0f5d4fb69.gtsb.io/kabel
https://deploy-preview-48--stevens-fp.netlify.app/kabel
https://f19f2f30.faculty-profiles.pages.dev/kabel

**Comment by mf-stevens** [comment]
Closing before merge because this site has been decommissioned.

**Pull request closed without merging**